plugin/keyring/common/system_keys_container.h

```diff
@@ -0,0 +1,63 @@
+#ifndef SYSTEM_KEYS_CONTAINER_INCLUDED
+#define SYSTEM_KEYS_CONTAINER_INCLUDED
+
+#include <my_global.h>
+#include <boost/move/unique_ptr.hpp>
+#include <map>
+#include "i_system_keys_container.h"
+#include "system_key_adapter.h"
+#include "logger.h"
+
+namespace keyring {
+
+// System_keys_container stores system keys together with links to their latest versions.
+// Latest version of a key itself is stored somewhere else (pressumably in keys_container).
+// For instance system_keys_container maps percona_binlog to its latest version percona_binlog:12,
+// the key percona_binlog:12 itself is stored in keys_container. System_key_container
+// only maps system key percona_binlog to its latest version, i.e. percona_binlog:12,
+// it does not know about other versions of the key. The keys returned by System_keys_container
+// are encapsulated in System_key_adapter which allows to retrieve information like key's
+// version.
+class System_keys_container : public ISystem_keys_container
+{
+public:
+  System_keys_container(ILogger *logger)
+    : logger(logger)
+  {}
+  ~System_keys_container();
+
+  // Returns key with latest version when called with plain system key (ex. percona_binlog)
```

Comment:
Maybe we should use Doxygen style comments?

Reply:
Done